packages/daimo-api/src/server/router.ts

```diff
@@ -624,6 +635,90 @@ export function createRouter(
         return status.isValid;
       }),
 
+    onAccountUpdate: publicProcedure
+      .input(
+        z.object({
+          address: zAddress,
+          inviteCode: z.string().optional(),
+          sinceBlockNum: z.number(),
+        })
+      )
+      .subscription(async (opts) => {
+        const { address, inviteCode } = opts.input;
+        // how often to send updates regardless of new transfers
+        // useful to update exchange rates and others.
+        const refreshInterval = 10_000;
+
+        return observable<AccountHistoryResult>((emit) => {
+          let lastEmittedBlock = opts.input.sinceBlockNum;
+          let getAccountHistoryPromise: Promise<AccountHistoryResult> | null =
+            null;
+
+          const pushHistory = (onlyOnNewTransfers: boolean) => {
+            getAccountHistoryPromise = getAccountHistory(
+              opts.ctx,
+              address,
+              inviteCode,
+              lastEmittedBlock,
+              watcher,
+              vc,
+              homeCoinIndexer,
+              ethIndexer,
+              foreignCoinIndexer,
+              profileCache,
+              noteIndexer,
+              reqIndexer,
+              inviteCodeTracker,
+              inviteGraph,
+              nameReg,
+              keyReg,
+              paymaster,
+              db
+            );
+
+            getAccountHistoryPromise
+              .then((history) => {
+                // we can have concurrent requests. discard those that arrived too late
+                if (history.lastBlock < lastEmittedBlock) {
+                  return;
+                }
+
+                if (onlyOnNewTransfers && history.transferLogs.length === 0) {
```

Comment:
what if, say, a keyAdded event arrives > we resync. no new transfers, but we have a new key.

won't this line discard the sync result?

Reply:
That's why there is interval sync on top of event-triggered sync that updates keys, exchange rates, linked accounts, named accounts, suggested actions, invites, notifications requests, and proposed swaps.

This way transfers are pushed in real time transfers while maintaining existing pull behavior for updating full state without major refactor.